src/requestLooper.ts

```diff
@@ -0,0 +1,119 @@
+import { DataFrame, DataQuery, DataQueryRequest, DataQueryResponse, LoadingState } from '@grafana/data';
```

Comment:
This file was basically taken from the timestream datasource with some modifications. In theory I think it should be shareable with Athena and other datasources that want to implement an async query flow. I haven't put it somewhere that's shared mainly for 2 reasons. 1) I'm not sure where it would best fit and 2) I wanted to try it out in different plugins first before trying to abstract it out into a shared library.